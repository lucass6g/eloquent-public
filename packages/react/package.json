{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
