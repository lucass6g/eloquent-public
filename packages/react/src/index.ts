export {Switch, Case, Default} from './components/conditional-switch'