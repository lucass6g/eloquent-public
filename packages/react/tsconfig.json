{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
