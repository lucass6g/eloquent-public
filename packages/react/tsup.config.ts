import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})