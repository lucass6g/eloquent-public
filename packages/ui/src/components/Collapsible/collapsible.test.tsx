import { fireEvent, render, RenderResult, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Collapsible } from '.';

const TRIGGER_TEXT = '@peduarte marcou 3 reposit√≥rios com estrela';
const CONTENT_TEXT = '@eloquent/tokens';


const CollapsibleTest = (props: React.ComponentProps<typeof Collapsible.Root>) => (
    <Collapsible.Root
        {...props}
    >
        <Collapsible.Trigger title={TRIGGER_TEXT} open={props.open} />
        <div data-content>
            @eloquent/react
        </div>
        <Collapsible.Content className='w-[400px]'>
            <div>
                {CONTENT_TEXT}
            </div>
        </Collapsible.Content>
    </Collapsible.Root >
);

describe('Collapsible component', () => {
    let rendered: RenderResult;
    let trigger: HTMLElement;
    let content: HTMLElement | null;

    beforeEach(() => {
        rendered = render(<CollapsibleTest />);
        trigger = rendered.getByText(TRIGGER_TEXT);
    });

    it('should render the Collapsible component and toggle content visibility', () => {
        expect(screen.getByText(TRIGGER_TEXT)).toBeInTheDocument();
        fireEvent.click(trigger);
        expect(screen.getByText(CONTENT_TEXT)).toBeInTheDocument();
        expect(rendered.baseElement).toMatchSnapshot();
    });

    describe('when clicking the trigger', () => {
        beforeEach(async () => {
            fireEvent.click(trigger);
            content = rendered.queryByText(CONTENT_TEXT);
        });

        it('should open the content', () => {
            expect(content).toBeVisible();
        });

        describe('and clicking the trigger again', () => {
            beforeEach(() => {
                fireEvent.click(trigger);
            });

            it('should close the content', () => {
                expect(content).not.toBeVisible();
            });
        });
    });
});