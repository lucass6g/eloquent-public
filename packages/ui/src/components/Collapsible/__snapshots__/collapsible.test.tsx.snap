// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Collapsible component > should render the Collapsible component and toggle content visibility 1`] = `
<body>
  <div>
    <div
      class="mb-1 space-y-2 [&>[data-disabled]]:opacity-50"
      data-state="open"
    >
      <button
        aria-controls="radix-:r0:"
        aria-expanded="true"
        class=""
        data-state="open"
        type="button"
      >
        <div
          class="ml-4 flex items-center justify-between space-x-4 [&>button>svg]:text-green-dark-500 [&>button>svg]:w-[16px] [&>button>svg]:h-[16px]"
        >
          <h4
            class="text-sm text-left font-bold text-green-dark-500"
          >
            @peduarte marcou 3 reposit√≥rios com estrela
          </h4>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm font-medium antialiased leading-4 p-0 m-0 border-none bg-transparent shadow-none text-secondary-foreground hover:shadow-none hover:bg-transparent disabled:bg-transparent disabled:shadow-none"
          >
            <svg
              class="lucide lucide-chevrons-up-down"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m7 15 5 5 5-5"
              />
              <path
                d="m7 9 5-5 5 5"
              />
            </svg>
          </button>
        </div>
      </button>
      <div
        data-content="true"
      >
        @eloquent/react
      </div>
      <div
        class="space-y-2 [&>div]:rounded-[6px] [&>div]:border [&>div]:px-4 [&>div]:py-3 [&>div]:text-sm [&>div]:text-neutral-500 w-[400px]"
        data-state="open"
        id="radix-:r0:"
        style=""
      >
        <div>
          @eloquent/tokens
        </div>
      </div>
    </div>
  </div>
</body>
`;
